### ISSUE TITLE
对比学习的损失权重

### ISSUE THREAD

**Issue by NJNUCS** [Haochen-Wang409/U2PL#52]
你好，我在问题区中了解到您的对比学习损失权重是0.1，但是我在代码里面看到的是损失权重是1，contra_loss = (contra_loss/ world_size* cfg["trainer"]["contrastive"].get("loss_weight", 1)，配置文件中并没有loss_weight

**Comment by Haochen-Wang409** [comment]
这个权重可以根据不同的数据集进行调整。您是否能复现我们的结果？

**Comment by NJNUCS** [comment]
对，我目前复现在voc2012数据集上的结果，不知道权重应该设置为多少合适



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Haochen ***@***.***&gt;; 
发送时间: 2022年6月12日(星期天) 下午4:42
收件人: ***@***.***&gt;; 
抄送: "완 \(ˆoˆ ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Haochen-Wang409/U2PL] 对比学习的损失权重 (Issue #52)





 
这个权重可以根据不同的数据集进行调整。您是否能复现我们的结果？

**Comment by Haochen-Wang409** [comment]
您可以尝试将权重设置为 0.1

根据我们的经验，U2PL 对这个参数较为鲁棒

**Comment by NJNUCS** [comment]
好的，谢谢你




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Haochen ***@***.***&gt;; 
发送时间: 2022年6月12日(星期天) 下午4:49
收件人: ***@***.***&gt;; 
抄送: "완 \(ˆoˆ ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Haochen-Wang409/U2PL] 对比学习的损失权重 (Issue #52)





 
您可以尝试将权重设置为 0.1
 
根据我们的经验，U2PL 对这个参数较为鲁棒

**Comment by NJNUCS** [comment]
不好意思，又来叨扰你了，low_mask_all = torch.cat(  ((label_l.unsqueeze(1) != 255).float(),low_entropy_mask.unsqueeze(1), ) )产生的mask为什么还要把label_l给拼在一起，以及后面的low_valid_pixel = torch.cat((label_l, label_u), dim=0) * low_mask_all,这样做，label_l是不是多余了

**Comment by Haochen-Wang409** [comment]
label_l 需要在计算对比学习 loss 时使用
https://github.com/Haochen-Wang409/U2PL/blob/main/train_semi.py#L483-L496

**Comment by Haochen-Wang409** [comment]
针对有标签数据，如果它的标签不是 ignore (255)，则把它作为一个 reliable pixel (即 low entropy)
针对无标签数据，需要它预测的概率分布熵较低，才能作为可靠的预测

**Comment by NJNUCS** [comment]
好滴，谢谢
